File: Data Structures & Algorithms/binary-tree-maximum-path-sum/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    int maxPathSum(TreeNode* root) {
        int res = root->val;
        dfs(root, &res);
        return res;
    }
    int dfs(TreeNode * root, int * res){
        if(!root) return 0;
        auto leftmax = max(dfs(root->left, res), 0);
        auto rightmax = max(dfs(root->right, res), 0);
        *res = max(root->val + leftmax + rightmax, *res);
        return root->val + max(leftmax, rightmax); 
    }
};

File: Data Structures & Algorithms/sqrtx/submission-1.py
class Solution:
    def mySqrt(self, x: int) -> int:
        l, r = 1, x
        res = 0
        while l <= r:
            mid = (l+r) //2
            if mid**2 == x:
                return mid
            if mid ** 2 < x:
                res = mid
                l = mid + 1
            else:
                r= mid -1
        return res
